frontend/src/redux/actions/actions.tsx

```diff
@@ -20,6 +20,15 @@ export const UI_APP_BAR_SET_ACTION = 'UI_APP_BAR_SET_ACTION';
 export const UI_THEME_SET = 'UI_THEME_SET';
 export const UI_RESET_PLUGIN_VIEWS = 'UI_RESET_PLUGIN_VIEWS';
 export const UI_PLUGINS_LOADED = 'UI_PLUGINS_LOADED';
+export const UI_BRANDING_APP_LOGO_COMPONENT = 'UI_BRANDING_APP_LOGO_COMPONENT';
+
+export interface BrandingProps {
+  logo: React.ComponentType<{
+    logoType: 'small' | 'large';
+    themeName: string;
+    [key: string]: any;
+  }> | null;
```

Comment:
I still find this a bit breaking from the current way we've used, but we probably require a full review into how to unify and make sure the API is as future proof as possible.
If @illume is fine with this, let's let it through.